### ISSUE TITLE
.NET Core February Update - 1.0.14, 1.1.11, 2.1.8 and 2.2.2

### ISSUE THREAD

**Issue by leecow** [dotnet/core#2304]
Included fixes address security issues detailed in the release notes.

## .NET Core

* **.NET Core 1.0.14 and .NET Core SDK 1.1.12**
  [Release Notes](https://github.com/dotnet/core/blob/master/release-notes/1.0/1.0.14/1.0.14.md)
  [Download 1.0](https://dotnet.microsoft.com/download/dotnet-core/1.0)

* **.NET Core 1.1.11 and .NET Core SDK 1.1.12**
  [Release Notes](https://github.com/dotnet/core/blob/master/release-notes/1.1/1.1.11/1.1.11.md)
  [Download 1.1](https://dotnet.microsoft.com/download/dotnet-core/1.1)

* **.NET Core 2.1.8 and .NET Core SDK 2.1.504**
  [Release Notes](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.8/2.1.8.md)
  [Download 2.1](https://dotnet.microsoft.com/download/dotnet-core/2.1)

* **.NET Core 2.2.2 and .NET Core SDK 2.2.104**
  [Release Notes](https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2.2/2.2.2.md)
  [Download 2.2](https://dotnet.microsoft.com/download/dotnet-core/2.2)

Please report any issues you find with these releases, either responding to this issue, creating a new issue here or creating a new issue in one of the following repos:

* [aspnet/Home](https://github.com/aspnet/Home/issues) - for ASP.NET Core questions
* [aspnet/EntityFrameworkCore](https://github.com/aspnet/EntityFrameworkCore/issues) - for Entity Framework Core questions
* [dotnet/cli](https://github.com/dotnet/cli/issues) - for CLI tools and questions
* [dotnet/corefx](https://github.com/dotnet/corefx/issues) - for API issues and questions
* [dotnet/coreclr](https://github.com/dotnet/coreclr/issues) - for runtime issues
* [dotnet/core-setup](https://github.com/dotnet/core-setup/issues) - for installer issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

## .NET Support Policies

Microsoft support policies are defined in the following documents:

* [Microsoft .NET Support Policy](https://www.microsoft.com/net/support/policy)
* [.NET Core OS Lifecycle Policy](https://github.com/dotnet/core/blob/master/os-lifecycle-policy.md)

**Comment by tjeason** [comment]
On Ubuntu 16.04, _aspnetcore-runtime_ 2.1.8 is missing for the Debian apt package repository. 

```
ubuntu$: wget -q https://packages.microsoft.com/config/ubuntu/16.04/packages-microsoft-prod.deb
ubuntu$: sudo dpkg -i packages-microsoft-prod.deb

ubuntu$: sudo apt-get install apt-transport-https
ubuntu$: sudo apt-get update
ubuntu$: sudo apt-get install dotnet-sdk-2.1

Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 dotnet-sdk-2.1 : Depends: aspnetcore-runtime-2.1 (>= 2.1.8) but 2.1.7-1 is to be installed
E: Unable to correct problems, you have held broken packages.
```
This is failing for the 2.1.8 .Net Core Release.

**Comment by tjeason** [comment]
This missing _aspnetcore-runtime_ for 2.1.8 release is resolved for Ubuntu 16.04. Mentioned in https://blogs.msdn.microsoft.com/dotnet/2019/02/12/net-core-february-2019/#comment-741405

**Comment by Rutix** [comment]
@leecow Some links in the opening post are wrong. The link for .NET Core 2.1.8 release notes should be: https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.8/2.1.8.md and not https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.7/2.1.8.md

And for .NET 2.2.2:
https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2.2/2.2.2.md and not https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2.1/2.2.2.md

**Comment by leecow** [comment]
Thanks @Rutix - fixed.

**Comment by PolitovArtyom** [comment]
Installing 2.1.8 hosting bundle or SDK breaks AspNetCore 2.1.7 apps

Application: dotnet.exe
CoreCLR Version: 4.6.27317.3
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileLoadException: Could not load file or assembly 'Microsoft.AspNetCore, Version=2.1.7.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

**Comment by kieronlanning** [comment]
Using 2.2.104 SDK in Azure DevOps .NET Core Tool Installer fails because it's not in the releases.csv. 

When will that be added?

**Comment by leecow** [comment]
@muratg and @joeloff - can you look at @PolitovArtyom 's issue above (https://github.com/dotnet/core/issues/2304#issuecomment-463154664)?

**Comment by leecow** [comment]
@kieronlanning - I see it in line 5 of the csv? If you're referring to the exe installer, those have not typically been included in the csv (though I see a few releases where they snuck in). 

@vivmishra

**Comment by kieronlanning** [comment]
@leecow Interesting, when I set my project to 2.2.104, the .NET Core SDK Installer fails:

```
2019-02-13T23:57:38.7785035Z Tool to install: .NET Core sdk version 2.2.104.
2019-02-13T23:57:38.7836944Z [command]/home/b33rdy/ado-agent/_work/_tasks/DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6/0.2.1/externals/get-os-distro.sh
2019-02-13T23:57:38.8548504Z Primary:linux-x64
2019-02-13T23:57:38.8549556Z Legacy:ubuntu.18.04
2019-02-13T23:57:38.8550836Z Detected platform (Primary): linux-x64
2019-02-13T23:57:38.8551637Z Detected platform (Legacy): ubuntu.18.04
2019-02-13T23:57:38.8553300Z Checking if a cached copy exists for this version...
2019-02-13T23:57:38.8566630Z The cache does not contain the requested version of .NET Core. Downloading and installing it now.
2019-02-13T23:57:38.8567272Z Getting URL to download .NET Core sdk version: 2.2.104.
2019-02-13T23:57:39.1111444Z ##[error]Cannot read property 'trim' of undefined
```

**Comment by leecow** [comment]
I suspect the Azure folks have not finished their deployment yet.

**Comment by muratg** [comment]
@PolitovArtyom we are tracking that issue here: https://github.com/aspnet/AspNetCore/issues/7546

**Comment by kieronlanning** [comment]
@leecow Is there a way to track the coordination between the various places - Azure AppServices, Azure Blob Deployments, GitHub's releases.csv etc?

A means to track it would also help in situations like this too. We would have checked a tracker, and just waited rather than commenting here...win/ win! 

We've been waiting for 2.2.2 for several months as it solves a problem with the `HttpContextAccessor`, so keen to get in updated ASAP.

**Comment by kieronlanning** [comment]
@leecow Looks like it's a problem with naming. It's referenced here: https://github.com/Microsoft/azure-pipelines-tasks/issues/9366

**Comment by bschapendonk** [comment]
@leecow 

Regarding the fix for the Azure DevOps .NET Core Tool Installer, according to https://github.com/Microsoft/azure-pipelines-tasks/issues/9366#issuecomment-456711057 only the error message is improved. 

So instead  of a `Cannot read property 'trim' of undefined` we now get a `The specified version's download links are not correctly formed in the supported versions document`.

**Comment by leecow** [comment]
Thanks @bschapendonk ... looks like this my fault :-/ Resolved 2.2.2 field names with https://github.com/dotnet/core/pull/2313.

**Comment by kieronlanning** [comment]
@leecow Amazing work, thanks! All our builds working a-ok now! Cheers!

**Comment by IanKemp** [comment]
Why was `Mvc.Formatters.Json.JsonInputFormatter` moved to `Mvc.NewtonsoftJson.NewtonsoftJsonInputFormatter.cs`?

**Comment by dougbu** [comment]
> Why was `Microsoft.AspNetCore.Mvc.Formatters.JsonInputFormatter` moved/renamed to `Microsoft.AspNetCore.Mvc.Formatters.NewtonsoftJsonInputFormatter.cs`?

@IanKemp moving the Json.NET dependencies is an intentional breaking change in the 3.0 preview releases. This and related changes were announced in aspnet/Announcements#325.

**Comment by BeigeBadger** [comment]
I am trying to use Azure DevOp's .NET Core SDK Installer step to install version 2.2.2 but I am running into an error with the download links/blobs not existing anymore.

Output:

`
##[section]Starting: Use .NET Core sdk 2.2.2
...
Task         : .NET Core SDK Installer
Description  : Acquires a specific version of the .NET Core SDK from internet or the local cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks.
Version      : 0.2.2
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?linkid=853651)
==============================================================================`
`Tool to install: .NET Core sdk version 2.2.2.
[command]C:\windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "& 'd:\a\_tasks\DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6\0.2.2\externals\get-os-platform.ps1'"
Primary:win-x64
Detected platform (Primary): win-x64
Checking if a cached copy exists for this version...
The cache does not contain the requested version of .NET Core. Downloading and installing it now.
Getting URL to download .NET Core sdk version: 2.2.2.
Could not fetch download information for version 2.2.2. Please check if the version specified is correct. You can refer the link for supported versions => https://github.com/dotnet/core/blob/master/release-notes/releases.json. Falling back to creating convention based URL.
[command]C:\windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "& 'd:\a\_tasks\DotNetCoreInstaller_b0ce7256-7898-45d3-9cb5-176b752bfea6\0.2.2\externals\install-dotnet.ps1' -Version 2.2.2 -DryRun"
dotnet-install: Payload URLs:
dotnet-install: Primary - https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.2/dotnet-sdk-2.2.2-win-x64.zip
dotnet-install: Legacy - https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.2/dotnet-dev-win-x64.2.2.2.zip
dotnet-install: Repeatable invocation: .\install-dotnet.ps1 -Version 2.2.2 -Channel LTS -Architecture x64 -InstallDir <auto>
Downloading: https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.2/dotnet-sdk-2.2.2-win-x64.zip
##[warning]Could not download installation package from this URL: https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.2/dotnet-sdk-2.2.2-win-x64.zip Error: {}
Downloading: https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.2/dotnet-dev-win-x64.2.2.2.zip
##[warning]Could not download installation package from this URL: https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.2/dotnet-dev-win-x64.2.2.2.zip Error: {}
##[error]Failed to download package for installation
##[section]Finishing: Use .NET Core sdk 2.2.2`

When I manually visit the links I get the following message: 

`<Error>
<Code>BlobNotFound</Code>
<Message>
The specified blob does not exist. RequestId:d2d1a17e-801e-0114-115d-d42dfa000000 Time:2019-03-06T20:43:25.7517795Z
</Message>
</Error>`

Changing the version value to use the `version-sdk` value from [this list](https://github.com/dotnet/core/blob/master/release-notes/releases.json) (2.2.104 - which I'm not sure is what I want) will successfully download the SDK but then it wants to use NuGet version >= 4.7 during the NuGet restore step (instead of v 4.4.1 which I have specified). I see there is an option to always use the latest version of NuGet but I would like to avoid this due to potential random build failures it could introduce in the future when versions change. 

Error message:

`##[error]The nuget command failed with exit code(1) and error(NU1102: Unable to find package NuGet.Frameworks with version (>= 4.7.0)  - Found 33 version(s) in NuGetOrg [ Nearest version: 4.7.0-preview1-4986 ]`

I would also rather not use a preview version of NuGet.

**Comment by karelz** [comment]
Closing as newer updates shipped - see #2432

**Comment by mswilson4040** [comment]
@BeigeBadger's solution worked for me. Couldn't get any 2.x version to work but 2.2.104 worked for whatever reason. According to the documentation, I should have been able to do 2.2.x or even 2.x but neither worked.
